README.md

```diff
@@ -121,6 +121,251 @@ curl https://minikube/atlas-contacts-app/v1/contacts -d '{"first_name": "Bob", "
 curl https://minikube/atlas-contacts-app/v1/contacts?_filter='first_name=="Mike"'
 ```
 
+##### Pagination
+
+Contacts App implements pagination in by adding application specific page token implementation.
+
+Actually the service supports "composite" pagination in a specific way:
+
+- limit and offset are still supported but without page token
+
+- if an user requests page token and provides limit then limit value will be used as a step for all further requests
+		`page_toke = null & limit = 2 -> page_token=base64(offset=2:limit=2)`
+		
+- if an user requests page token and provides offset then only first time the provided offset is applied
+		`page_token = null & offset = 2 & limit = 2 -> page_token=base64(offset=4:limit=2)`
+
+```shell
+GET http://localhost:8080/atlas-contacts-app/v1/contacts
+
+HTTP/1.1 200 OK
+Content-Type: application/json
+Grpc-Metadata-Content-Type: application/grpc
+Date: Fri, 25 May 2018 16:40:12 GMT
+Content-Length: 603
+
+{
+  "results": [
+    {
+      "emails": [
+        {
+          "address": "one@mail.com",
+          "id": "1"
+        }
+      ],
+      "first_name": "Mike",
+      "id": "1",
+      "primary_email": "one@mail.com"
+    },
+    {
+      "emails": [
```

Comment:
I think it would be fine to truncate this (and all subsequent) output with a comment like `... followed by <n> more contacts` to help shorten the document

Reply:
I was thinking to make it shorter, but left it as it is just in case. I will cut it.